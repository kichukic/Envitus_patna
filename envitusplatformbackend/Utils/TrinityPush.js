const request = require('request');

var Trinity = module.exports = {
  accessToken: '',
  refreshToken: '',
  auth: async function () {
    const options = {
      'method': 'POST',
      'url': process.env.PUSH_TRINITY_URL + '/HTTPProtAdaptorService/oauth/token',
      'headers': {
        'Authorization': 'Basic ' + process.env.TRINITY_APP_AUTH
      },
      formData: {
        'username': process.env.TRINITY_USERNAME,
        'password': process.env.TRINITY_PASSWORD,
        'grant_type': 'password'
      }
    };

    var resp = await new Promise(resolve => {
      request(options, function (error, response) {
        if (error) { resolve(false) }
        else { resolve(JSON.parse(response.body)) }
      });
    });

    
    if (resp) {
      Trinity.accessToken = resp.access_token;
      Trinity.refreshToken = resp.refresh_token;
      setTimeout(Trinity.refresh, ((resp.expires_in - (15 * 60)) * 1000));
    }
  },
  refresh: async function () {
    const options = {
      'method': 'POST',
      'url': process.env.PUSH_TRINITY_URL + '/HTTPProtAdaptorService/oauth/token',
      'headers': {
        'Authorization': 'Basic ' + process.env.TRINITY_APP_AUTH
      },
      formData: {
        'username': process.env.TRINITY_USERNAME,
        'password': process.env.TRINITY_PASSWORD,
        'grant_type': 'password',
        'refresh_token': Trinity.refreshToken
      }
    };

    var resp = await new Promise(resolve => {
      request(options, function (error, response) {
        if (error) { resolve(false) }
        else { resolve(JSON.parse(response.body)) }
      });
    });

    if (resp) {
      Trinity.accessToken = resp.access_token;
      Trinity.refreshToken = resp.refresh_token;
      setTimeout(Trinity.refresh, ((resp.expires_in - (15 * 60)) * 1000));
    }
  }
}
