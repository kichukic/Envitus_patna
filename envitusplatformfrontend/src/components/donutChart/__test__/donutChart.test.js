import React from 'react';
import { shallow } from 'enzyme';
import DonutChart from './../donutChart';
import { findByDataTest } from '../../../Utils';

const preConfig = (props={}) => {
    const component = shallow(<DonutChart {...props} />);
    return component;
};

describe('DonutChart Test', () => {
    it('snapshot', () => {
        const wrapper = shallow(<DonutChart/>);   
        expect(wrapper).toMatchSnapshot();
    });

    it('Redering Properly with prop values', () => {
        let component;
        component = preConfig();
        const wrapper = findByDataTest(component, 'semiCircleComponent');
        expect(wrapper.length).toBe(1);
    });

    it('draws Chart', () => {
        let component;
        const props = {
            value: 20,
            max: 50,
            min: 2,
            unit: "mg",
            color: "F5F5F5"
        };
        component = preConfig(props);
        component.setState({
            node: React.createRef()
        });
        expect(component.instance().render().ref.current).toBe(null);  
    })
});