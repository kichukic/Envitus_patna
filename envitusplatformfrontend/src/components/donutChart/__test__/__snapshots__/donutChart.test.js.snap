// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart Test snapshot 1`] = `
<svg
  data-test="donutChartComponent"
/>
`;
